### ISSUE TITLE
Improvement of NSAttributedString extension's 'applying' function

### ISSUE THREAD

**Issue by DimaZava** [SwifterSwift/SwifterSwift#777]
I propose to add 'options' as default parameter to function 'apply' in NSAttributedStringExtensions file, line 105. For example there's common case of using .caseInsensitive option which cannot be ignored

```
/// SwifterSwift: Apply attributes to substrings matching a regular expression
///
/// - Parameters:
///   - attributes: Dictionary of attributes
///   - pattern: a regular expression to target
///   - options: The regular expression options that are applied to the expression during matching. See NSRegularExpression.Options for possible values.
/// - Returns: An NSAttributedString with attributes applied to substrings matching the pattern
func applying(attributes: [NSAttributedString.Key: Any], toRangesMatching pattern: String, options: NSRegularExpression.Options = []) -> NSAttributedString {
    guard let pattern = try? NSRegularExpression(pattern: pattern, options: options]) else { return self }

    let matches = pattern.matches(in: string, options: [], range: NSRange(0..<length))
    let result = NSMutableAttributedString(attributedString: self)

    for match in matches {
        result.addAttributes(attributes, range: match.range)
    }

    return result
}
```

**Comment by omaralbeik** [comment]
Would you please open a PR for this 🚀

**Comment by LucianoPAlmeida** [comment]
Fixed by #791 Thanks @DimaZava :))
